### PR TITLE
Add a create method with context and namePrefix to ActorMaterialize‘s Java API.

### PR DESCRIPTION
**Pull request by He-Pin**
Otherwise I need to pass the first parameter a `null`

### CODE REVIEW

**Review by johanandren — APPROVED** [review]
LGTM, but needs reformatting

File: akka-stream/src/main/scala/akka/stream/ActorMaterializer.scala [hunk 1]
```diff
@@ -123,6 +123,21 @@ object ActorMaterializer {
...
+   * The required [[akka.actor.ActorRefFactory]]
+   * (which can be either an [[akka.actor.ActorSystem]] or an [[akka.actor.ActorContext]])
+   * will be used to create these actors, therefore it is *forbidden* to pass this object
+   * to another actor if the factory is an ActorContext.
+   *
+   * The `namePrefix` is used as the first part of the names of the actors running
+   * the processing steps. The default `namePrefix` is `"flow"`. The actor names are built up of
+   * `namePrefix-flowNumber-flowStepNumber-stepName`.
+   */
+  def create(context: ActorRefFactory, namePrefix: String): ActorMaterializer =
```
Comment (patriknw) [inline akka-stream/src/main/scala/akka/stream/ActorMaterializer.scala]: then it should also be added for Scala API:
```
def apply(materializerSettings: ActorMaterializerSettings, namePrefix: String)(implicit context: ActorRefFactory): ActorMaterializer =
```

**Comment by hepin1989** [comment]
I will do this tomorrow.

File: akka-stream/src/main/scala/akka/stream/ActorMaterializer.scala (hunk 1 above)
Comment (hepin1989) [reply akka-stream/src/main/scala/akka/stream/ActorMaterializer.scala]: I have chaged it.

**Comment by hepin1989** [comment]
@patriknw Do you have any time :)

**Comment by raboof** [comment]
I don't see the apply method mentioned in https://github.com/akka/akka/pull/26526#discussion_r265899495 ?

**Review by patriknw — APPROVED** [review]
LGTM

File: akka-stream/src/main/scala/akka/stream/ActorMaterializer.scala (hunk 1 above)
Comment (patriknw) [reply akka-stream/src/main/scala/akka/stream/ActorMaterializer.scala]: I think I was wrong here. There is already one that can be used in Scala:
```
  def apply(materializerSettings: Option[ActorMaterializerSettings] = None, namePrefix: Option[String] = None)(
      implicit context: ActorRefFactory): ActorMaterializer 
```

**Pull request merged**